include/gatekeeper_gk.h

```diff
@@ -45,14 +45,32 @@ enum gk_fib_action {
 	/* Forward the packet to the corresponding Grantor. */
 	GK_FWD_GRANTOR,
 
-	/* Forward the packet to the corresponding gateway. */
-	GK_FWD_GATEWAY,
+	/*
+	 * Forward the packet to the corresponding gateway
+	 * in the front network.
+	 */
+	GK_FWD_GATEWAY_FRONT_NET,
+
+	/*
+	 * Forward the packet to the corresponding gateway
+	 * in the back network.
+	 */
+	GK_FWD_GATEWAY_BACK_NET,
+
+	/*
+	 * The destination address is a neighbor in the front network.
+	 * Forward the packet to the destination directly.
+	 */
+	GK_FWD_NEIGHBOR_FRONT_NET,
 
 	/*
-	 * The destination address is a neighbor.
+	 * The destination address is a neighbor in the back network.
 	 * Forward the packet to the destination directly.
 	 */
-	GK_FWD_NEIGHBOR,
+	GK_FWD_NEIGHBOR_BACK_NET,
+
+	/* Forward the packet to the front interface. */
+	GK_FWD_FRONT_NET,
```

Comment:
`GK_FWD_FRONT_NET` became redundant since it is exactly the same as having an LPM prefix pointing to a FIB entry whose action is either `GK_FWD_GATEWAY_FRONT_NET` or `GK_FWD_NEIGHBOR_FRONT_NET`. The only change will be in the code that edits the FIB. The analog is true for `GK_FWD_BACK_NET` (below).

Therefore, drop `GK_FWD_FRONT_NET` and `GK_FWD_BACK_NET`, and adjust the code accordingly.